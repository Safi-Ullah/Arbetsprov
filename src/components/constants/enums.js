export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    }
};