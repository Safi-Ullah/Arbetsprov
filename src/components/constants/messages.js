export const PromptMessages = {
    DELETE_TASK_PROMPT: `Are you sure you want to permanently delete this task?
        You will not be able to restore in the future.`
};