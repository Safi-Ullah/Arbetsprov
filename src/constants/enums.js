export const Enums = {
    TASK_STATUS: {
        created: 1,
        active: 2,
        completed: 3,
        onhold: 4,
        deleted: 5
    },
    TASK_ACTIONS: {
        edit: 1,
        setComplexity: 2,
        delete: 3,
    },
    APP_SCREENS: {
        dashboard: 1,
        deletedTasks: 2
    },
    SORT_BY_COMPLEXITY: {
        ascending: 1,
        descending: 2
    },
    TEST_TYPES: {
        dom: 'Dom test',
        snapshot: 'Snapshot test'
    }
};