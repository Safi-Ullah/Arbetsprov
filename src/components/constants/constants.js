import { Enums } from './enums';

export const TASK_ACTION_HANDLERS = {
    [Enums.TASK_ACTIONS.edit]: 'displayEditTaskModal',
    [Enums.TASK_ACTIONS.delete]: 'displayConfirmationDialog'
};