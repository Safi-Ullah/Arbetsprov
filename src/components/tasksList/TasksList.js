import React from 'react';
import PropTypes from 'prop-types';
import { TaskListItem } from './TaskListItem';
import { TASK_NEXT_STATUS, Enums } from '../../constants';
import { getKeyFromValue } from '../../utils';

export const TasksList = React.memo(props => {

    const {
        tasks, displayConfirmationDialog,
        displayEditTaskModal, status, changeStatus
    } = props;

    let taskNextStatus = status in TASK_NEXT_STATUS
        ? TASK_NEXT_STATUS[status]
        : Object.keys(Enums.TASK_STATUS).map(key =>
            Enums.TASK_STATUS[key]).filter(taskStatus => taskStatus !== status);

    return <ul className="list-group">
        {
            tasks.length
                ? tasks.map((task, index) => <TaskListItem
                    displayConfirmationDialog={displayConfirmationDialog}
                    displayEditTaskModal={displayEditTaskModal}
                    task={task}
                    key={index}
                    changeStatus={changeStatus}
                    taskNextStatus={taskNextStatus} />
                )
                : <span className="text-center text-muted">
                    No tasks {getKeyFromValue(Enums.TASK_STATUS, status)}.
                </span>
        }
    </ul>
});

TasksList.propTypes = {
    tasks: PropTypes.array.isRequired,
    displayConfirmationDialog: PropTypes.func.isRequired,
    displayEditTaskModal: PropTypes.func.isRequired,
    status: PropTypes.number.isRequired,
    changeStatus: PropTypes.func.isRequired
};