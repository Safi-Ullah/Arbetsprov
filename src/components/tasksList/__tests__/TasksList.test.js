import React from 'react';
import TestRenderer from 'react-test-renderer';
import { TasksList } from '../TasksList';
import { Enums } from '../../../constants';

const props = {
    tasks: [],
    displayConfirmationDialog: jest.fn(),
    displayEditTaskModal: jest.fn(),
    status: Enums.TASK_STATUS.created,
    changeStatus: jest.fn()
};

describe(Enums.TEST_TYPES.snapshot, () => {
    it('show taskslist without any tasks', () => {
        const tree = TestRenderer.create(<TasksList {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('show taskslist with tasks', () => {
        props.tasks = [{
            title: 'test',
            desc: 'test',
            complexity: '0',
            assignee: 'test',
            status: Enums.TASK_STATUS.created
        }];
        const tree = TestRenderer.create(<TasksList {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});