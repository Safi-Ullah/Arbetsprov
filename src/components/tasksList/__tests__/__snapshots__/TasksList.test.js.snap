// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test show taskslist with tasks 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item"
  >
    <button
      className="btn btn-link text-break"
      onClick={[Function]}
    >
      test
    </button>
    <span
      className="badge badge-info"
    >
      0
    </span>
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="close ml-2"
      data-toggle="dropdown"
      id="dropdownMenuButton"
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ⁝
      </span>
    </button>
    <div
      aria-labelledby="dropdownMenuButton"
      className="dropdown-menu"
    >
      <button
        className="dropdown-item"
        onClick={[Function]}
      >
        Edit
      </button>
      <div
        className="dropdown-divider"
      />
      <span
        className="dropdown-subheading"
      >
        Change Status
      </span>
      <button
        className="dropdown-item text-capitalize"
        disabled={false}
        onClick={[Function]}
      >
        onhold
      </button>
    </div>
    <button
      className="close"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
        className="delete-btn"
      >
        ×
      </span>
    </button>
  </li>
</ul>
`;

exports[`Snapshot test show taskslist without any tasks 1`] = `
<ul
  className="list-group"
>
  <span
    className="text-center text-muted"
  >
    No tasks 
    created
    .
  </span>
</ul>
`;
